File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    var pro = progresses;
    var spd = speeds;
    var temp = [];
    
    for(let i = 0; i < pro.length; i++){
        temp[i] = Math.ceil((100 - pro[i])/spd[i]); 
    }
    var max = temp[0];
    var cnt = 1;
    
    console.log("temp", temp);
    for(let i = 1; i < temp.length; i++){
        console.log('max', max, 'temp', temp[i]);
        
        if(temp[i] > max){
            console.log('cnt', cnt);
            answer.push(cnt);
            max = temp[i];
            cnt = 0;
        }
        cnt++;
    }
    answer.push(cnt);

    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <iostream>
#include <cmath>
using namespace std;

vector<int> solution(vector<int> p, vector<int> s) {
    vector<int> answer;
    vector<int> temp;

    // 남은 작업량 계산
    for (int i = 0; i < p.size(); i++) {
        int result = ceil(static_cast<double>(100 - p[i]) / s[i]);
        temp.push_back(result);
    }

    // 배포 그룹 계산
    int max = temp.front();
    int cnt = 1;

    for (int i = 1; i < temp.size(); i++) {
        if (temp[i] > max) {
            max = temp[i];
            answer.push_back(cnt);
            cnt = 1;
        } else {
            cnt++;
        }
    }
    answer.push_back(cnt);

    return answer;
}

int main() {
    vector<int> p = {93, 30, 55};  // 진행률
    vector<int> s = {1, 30, 5};    // 속도
    vector<int> result = solution(p, s);

    for (int x : result) {
        cout << x << " ";
    }

    return 0;
}